flink-runtime/src/test/java/org/apache/flink/runtime/util/config/memory/ProcessMemoryUtilsTestBase.java

```diff
@@ -350,4 +368,35 @@ public MemorySize getTotalProcessMemorySize() {
 			throw new UnsupportedOperationException();
 		}
 	}
+
+	private static class DummyFlinkMemoryUtils implements FlinkMemoryUtils<DummyFlinkMemory> {
+		@Override
+		public DummyFlinkMemory deriveFromRequiredFineGrainedOptions(Configuration config) {
+			return new DummyFlinkMemory();
+		}
+
+		@Override
+		public DummyFlinkMemory deriveFromTotalFlinkMemory(Configuration config, MemorySize totalFlinkMemorySize) {
+			throw new UnsupportedOperationException();
+		}
+	}
+
+	private static class DummyFlinkMemory implements FlinkMemory {
+		private static final long serialVersionUID = 1L;
+
+		@Override
+		public MemorySize getJvmHeapMemorySize() {
+			throw new UnsupportedOperationException();
+		}
+
+		@Override
+		public MemorySize getJvmDirectMemorySize() {
+			throw new UnsupportedOperationException();
+		}
+
+		@Override
+		public MemorySize getTotalFlinkMemorySize() {
+			throw new UnsupportedOperationException();
+		}
+	}
```

Comment:
I don't really understand why we need these helper classes. Looking at the method bodies it looks as if they are not really needed. The fact that we need to provide them for the test is an indicator that there is something wrong with the code abstraction as the actual change is not easily testable.

Reply:
Indeed, the abstractions are not optimal now for the tests setup.
There is an issue planned for refactoring of memory utils tests: [FLINK-16754](https://issues.apache.org/jira/browse/FLINK-16754). I think we can make a better step towards it in this PR by abstracting only setting of the internal memory. I changed the approach.